lib/Onyx.js

```diff
@@ -49,6 +49,47 @@ let defaultKeyStates = {};
 // Connections can be made before `Onyx.init`. They would wait for this task before resolving
 const deferredInitTask = createDeferredTask();
 
+let batchUpdatesPromise = null;
+let batchUpdatesQueue = [];
+
+/**
+ * We are batching together onyx updates. This helps with use cases where we schedule onyx updates after each other.
+ * This happens for example in the Onyx.update function, where we process API responses that might contain a lot of
+ * update operations. Instead of calling the subscribers for each update operation, we batch them together which will
+ * cause react to schedule the updates at once instead of after each other. This is mainly a performance optimization.
+ * @returns {Promise}
+ */
+function maybeFlushBatchUpdates() {
+    if (batchUpdatesPromise) {
+        return batchUpdatesPromise;
+    }
+
+    batchUpdatesPromise = new Promise((resolve) => {
+        /* We use (setTimeout, 0) here which should be called once native module calls are flushed (usually at the end of the frame)
+         * We may investigate if (setTimeout, 1) (which in React Native is equal to requestAnimationFrame) works even better
+         * then the batch will be flushed on next frame.
+         */
+        setTimeout(() => {
+            const updatesCopy = batchUpdatesQueue;
+            batchUpdatesQueue = [];
+            batchUpdatesPromise = null;
+            unstable_batchedUpdates(() => {
+                updatesCopy.forEach((applyUpdates) => {
+                    applyUpdates();
+                });
+            });
+
+            resolve();
+        }, 0);
+    });
+    return batchUpdatesPromise;
+}
+
+function batchUpdates(updates) {
```

Comment:
```suggestion
function queueUpdates(updates) {
```

I think this is a clearer name for this function?

Reply:
but also it is not only queueing, it's queueing and maybe processing. Not sure. Maybe `batchUpdates` is explicative enough?

Reply:
This code was actually just moved and I didn't change it so maybe it would be better to keep it as is?

Reply:
Yeah, I didn't review the previous PR so I think it's NAB